spec/services/data_migrations/populate_application_choice_provider_ids_spec.rb

```diff
@@ -32,6 +32,8 @@ def expected_provider_ids(application_choice)
 
     expect(application_choices[3].reload.provider_ids.length).to eq(4)
     expected_arrays = application_choices.map { |a| expected_provider_ids(a) }
-    expect(ApplicationChoice.all.map(&:provider_ids)).to eq(expected_arrays)
+    actual_arrays = ApplicationChoice.order('created_at').all.map(&:provider_ids)
+
+    expect(expected_arrays & actual_arrays).to eq(expected_arrays) # order-insensitive
```

Comment:
Could you use [contain_exactly](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/contain-exactly-matcher)? 

```suggestion
    expect(ApplicationChoice.all.map(&:provider_ids)).to contain_exactly(expected_arrays) #order-insensitive
```

Reply:
I think this will need a splat `*` but may still be easier to read, will update